# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/ThreadEventTarget.cpp
# Commit: 95a6f05099b7
# Full Hash: 95a6f05099b72e4edabcf6cf6811ae1822c4900a
# Author: Karl Tomlinson <karlt+@karlt.net>
# Date: 2020-01-08 21:56:06
# Description:
#   Bug 1604005 implement ThreadEventTarget::IsOnCurrentThreadInfallible() to handle case of null mThread r=froydnj
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D58999
# ==============================================================================

diff -r 171648f3e48e -r 95a6f05099b7 xpcom/threads/ThreadEventTarget.cpp
--- a/xpcom/threads/ThreadEventTarget.cpp	Wed Jan 08 18:41:01 2020 +0000
+++ b/xpcom/threads/ThreadEventTarget.cpp	Wed Jan 08 15:25:27 2020 +0000
@@ -185,6 +185,8 @@
 
 NS_IMETHODIMP_(bool)
 ThreadEventTarget::IsOnCurrentThreadInfallible() {
-  // Rely on mThread being correct.
-  MOZ_CRASH("IsOnCurrentThreadInfallible should never be called on nsIThread");
+  // This method is only going to be called if `mThread` is null, which
+  // only happens when the thread has exited the event loop.  Therefore, when
+  // we are called, we can never be on this thread.
+  return false;
 }
